tests/publiccloud/img_proof.pm

```diff
@@ -92,6 +93,18 @@ sub run {
     # Because the IP address of instance might change during img_proof due to the hard-reboot, we need to re-add the ssh public keys
     assert_script_run(sprintf('ssh-keyscan %s >> ~/.ssh/known_hosts', $instance->public_ip));
 
+    if (is_hardened) {
+        # Add soft-failure for https://bugzilla.suse.com/show_bug.cgi?id=1220269
+        zypper_call "in jq";
+        my $outcome = script_output "cat $img_proof->{results} | jq '.tests[] | select(.nodeid | startswith(\"test_sles_hardened\")) | .outcome'";
+        if ($outcome =~ m/"failed"/) {
+            # Change "failed" to "passed"
+            assert_script_run "cat $img_proof->{results} | jq '.tests |= map(if (.nodeid | startswith(\"test_sles_hardened\")) then .outcome = \"passed\" else . end)' > tmp.json";
```

Comment:
In addition to above note, and for clarification only, being Not so clear to me what its scope, but, is there no other possible way to achieve that requirement, using more atomic perl or shell programming?

Reply:
I will do it in Perl. Shell is not the right tool to modify JSON.

Reply:
ok, then kind question for better view of the context:  
what specific parameter of that json structure shall be tweaked and  
why we needed to just change the full json in this case, which routine need that full properly-updated-json to pass tests ?

Reply:
Eventually, also some more comment for debugging help and future maintenance, could be added to explain it, at end of your good work

Reply:
I need to change the JSON in https://openqa.suse.de/tests/13892979/file/img_proof-20240327124422.results.

The structure looks like this and I changed the `outcome` from `failed` to `passed` only for the `nodeid` that starts with `test_sles_hardened.py`:

```
{
  "tests": [
...
    {
      "nodeid": "test_sles_hardened.py::test_sles_hardened[paramiko://52.212.166.248]",
      "lineno": 38,
      "outcome": "passed",
```